### PR TITLE
Retry nix commands if HTTP errors are encountered

### PR DESCRIPTION
**Pull request by KaiserKarel**

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [NX-35 Retry builds on network failure](https://app.clickup.com/t/20465559/NX-35)

**Review by aeryz — APPROVED** [review]

**Review by blasrodri — APPROVED** [review]

**Review by cor — CHANGES_REQUESTED** [review]
Best to fix sshellcheck issues (https://www.shellcheck.net/) see Slack for screenshot

**Comment by KaiserKarel** [comment]
> Best to fix shellcheck issues (https://www.shellcheck.net/) see Slack for screenshot

Done sir

**Review by cor — APPROVED** [review]

File: .github/templates/watch-exec/action.yml [hunk 1]
```diff
@@ -45,6 +45,38 @@ runs:
...
+        RETRIES=5               
+        CMD="nix -- run .#benchmarks-once-dali --no-update-lock-file --show-trace -L 2> >(tee -a $LOG_FILE >&2)" 
+        # Marker that a network error occurred.
+        BAD="HTTP error 200 ('')"  
+        LOG_FILE=/tmp/out.log
+
+        function run() {
+            eval "${CMD?}"
+            STATUS_CODE=$?
+            if [[ $? -ne 0 ]] ; then
```
Comment (obsessed-cake) [inline .github/templates/watch-exec/action.yml:60]: here `$?` will equal `0` always
```
$ STATUS_CODE=$?
$ echo $?
0
$ echo $STATUS_CODE
1
```

Comment (obsessed-cake) [inline .github/templates/watch-exec/action.yml:60]:
```suggestion
            if [[ $STATUS_CODE -ne 0 ]] ; then
```

Comment (KaiserKarel) [reply .github/templates/watch-exec/action.yml:60]: Good catch

File: tee [hunk 2]
```diff
@@ -0,0 +1,29 @@
+warning: Git tree '/workspaces/composable' is dirty
```
Comment (hussein-aitlahcen) [inline tee:1]: Do we keep this file?

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by obsessed-cake — APPROVED** [review]

**Pull request merged**